src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,36 +146,41 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
-                    var tokenSource = new CancellationTokenSource();
-
                     var receiveTask = Task.Run(async () =>
                     {
-                        try
-                        {
-                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, tokenSource, pipeline).ConfigureAwait(false);
-                            receiveCircuitBreaker.Success();
-                        }
-                        catch (MessageQueueException ex)
+                        using (var tokenSource = new CancellationTokenSource())
```

Comment:
We could, but not sure if this adds anything because then each strategy would need to deal with it. Scope creep ;) ?